kyaml/kio/kio_test.go

```diff
@@ -149,3 +149,62 @@ items:
 		t.FailNow()
 	}
 }
+
+func TestEmptyResult(t *testing.T) {
+	actual := &bytes.Buffer{}
+	output := ByteWriter{Writer: actual}
+
+	generatorFunc := FilterFunc(func(nodes []*yaml.RNode) ([]*yaml.RNode, error) {
+		nodes = append(nodes, yaml.NewMapRNode(&map[string]string{
+			"foo": "bar",
+		}))
+		return nodes, nil
+	})
+	emptyFunc := FilterFunc(func(nodes []*yaml.RNode) ([]*yaml.RNode, error) { return nodes, nil })
+
+	// Because ContinueOnEmptyResult is false by default, the pipeline will stop on the first filter
+	p := Pipeline{
+		Outputs: []Writer{output},
+		Filters: []Filter{emptyFunc, generatorFunc},
+	}
+
+	err := p.Execute()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !assert.Equal(t,
+		"", strings.TrimSpace(actual.String())) {
+		t.FailNow()
+	}
+}
+
+func TestContinueOnEmptyResult(t *testing.T) {
```

Comment:
Since these two tests have the same setup, please combine into 
one test driven by a loop over a map, e.g. https://github.com/kubernetes-sigs/kustomize/blob/master/api/filters/refvar/refvar_test.go#L17

```
TestContinueOnEmptyBehavior() {
   cases := map[string]struct {
                continueBehavior bool
                out string
        }{
           "quit on empty" : {continueB:false, out: ""},
           "continue on empty": {continueB:true, out: "foo:bar"},
    }
   for tn, tc := range cases {
    // setup out buffer, generator, etc. 
    // run pipe, test.
  }
```